packages/core/src/new-api/internal/new-execution/types/messages.ts

```diff
@@ -0,0 +1,96 @@
+import { SolidityParameterType } from "../../../types/module";
+
+import {
+  CallExecutionResult,
+  DeploymentExecutionResult,
+} from "./execution-result";
+import { DeploymentExecutionStateFutureTypes } from "./execution-state";
+import { Transaction, TransactionReceipt } from "./jsonrpc";
+import { NetworkInteraction } from "./network-interaction";
+
+export type JournalMessage =
+  | RunStartMessage
+  | DeploymentExecutionStateInitializeMessage
+  | DeploymentExecutionStateCompleteMessage
+  | CallExecutionStateInitializeMessage
+  | CallExecutionStateCompleteMessage
+  | NetworkInteractionRequestMessage
+  | TransactionSendMessage
+  | TransactionConfirmMessage;
+
+export enum JournalMessageType {
+  RUN_START = "RUN_START",
+  DEPLOYMENT_EXECUTION_STATE_INITIALIZE = "DEPLOYMENT_EXECUTION_STATE_INITIALIZE",
+  DEPLOYMENT_EXECUTION_STATE_COMPLETE = "DEPLOYMENT_EXECUTION_STATE_COMPLETE",
+  CALL_EXECUTION_STATE_INITIALIZE = "CALL_EXECUTION_STATE_INITIALIZE",
+  CALL_EXECUTION_STATE_COMPLETE = "CALL_EXECUTION_STATE_COMPLETE",
+  NETWORK_INTERACTION_REQUEST = "NETWORK_INTERACTION_REQUEST",
```

Comment:
I think we are missing these:

`TRANSACTION_SIMULATION_ERROR`
`TRANSACTION_DROPPED`
`TRANSACTION_REPLACED_BY_USER_INTERFERENCE`

Reply:
Oh, simulation errors are encoded as `DEPLOYMENT_EXECUTION_STATE_COMPLETE` and `CALL_EXECUTION_STATE_COMPLETE`, right?

Reply:
Don't we need a message for `StaticCall` results?